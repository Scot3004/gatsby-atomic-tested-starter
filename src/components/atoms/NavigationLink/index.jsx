import styled from 'styled-components';
import { Link } from 'gatsby';

const NavigationLink = styled(Link)`
  align-items: center;
  color: ${props => props.theme.textColor};
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  font-weight: bold;
  justify-content: center;
  padding: 0.5rem;
  text-decoration: none;

  &:visited {
    color: ${props => props.theme.textColor};
  }

  &.active {
    color: ${props => props.theme.visitedColor};
  }

  &:hover {
    color: ${props => props.theme.primaryColor};
  }

  &:before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    margin: -$border; /* !importanté */
    border-radius: inherit; /* !importanté */
    background: linear-gradient(to right, red, orange);
  }
`;

export default NavigationLink;
