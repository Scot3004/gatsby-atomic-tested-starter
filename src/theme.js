const theme = {
  backgroundColor: '#f5f5f5',
  textColor: '#000',
  mutedColor: '#555',
  primaryColor: '#663399',
  visitedColor: '#402060'
};

export default theme;
