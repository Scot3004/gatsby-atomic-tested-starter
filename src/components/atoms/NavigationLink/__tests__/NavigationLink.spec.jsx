import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import theme from '../../../../theme';
import NavigationLink from '..';

describe('When I see the navigation links', () => {
  const tree = renderer.create(<NavigationLink theme={theme} />).toJSON();
  test('Then they should look neutral if the page is not the current page', () => {
    expect(tree).toMatchSnapshot();
  });
});
