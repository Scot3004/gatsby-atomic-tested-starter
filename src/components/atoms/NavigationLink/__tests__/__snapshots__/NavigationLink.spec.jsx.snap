// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I see the navigation links Then they should look neutral if the page is not the current page 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #000;
}

.c0.active {
  color: #402060;
}

.c0:hover {
  color: #663399;
}

.c0:before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
  margin: -$border;
  border-radius: inherit;
  background: linear-gradient(to right,red,orange);
}

<a
  className="c0"
  theme={
    Object {
      "backgroundColor": "#f5f5f5",
      "mutedColor": "#555",
      "primaryColor": "#663399",
      "textColor": "#000",
      "visitedColor": "#402060",
    }
  }
/>
`;
